### PR TITLE
Add multi-thread unittest for mlflow handler

### PR DESCRIPTION
**Pull request by binliunls**
Signed-off-by: binliu <binliu@nvidia.com>
### Description

Add a multi-thread unit test for mlflow handler.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Review by Nic-Ma — APPROVED** [review]

File: tests/test_handler_mlflow.py [hunk 1]
```diff
@@ -61,6 +86,18 @@ def _update_metric(engine):
...
+        test_uri_list = ["monai_mlflow_test1", "monai_mlflow_test2"]
+        with ThreadPoolExecutor(2, "Training") as executor:
+            futures = {}
+            for t in test_uri_list:
+                futures[t] = executor.submit(dummy_train, t)
+
+            for _, future in futures.items():
+                res = future.result()
+                self.assertTrue(len(glob.glob(res)) > 0)
+                shutil.rmtree(res)
```
Comment (wyli) [inline tests/test_handler_mlflow.py:99]: will this leave unclosed temporary files/folders if the test fails?

Comment (Nic-Ma) [reply tests/test_handler_mlflow.py:99]: Hi @binliunls ,

Please refer to other test program to set the `tempdir`, for example:
https://github.com/Project-MONAI/MONAI/blob/dev/tests/test_lmdbdataset_dist.py#L34-L38

Thanks.

Comment (binliunls) [reply tests/test_handler_mlflow.py]: Hi @Nic-Ma ,
      I have added a try finally logic. Is it ok for this issue.

Thanks,
Bin

File: tests/test_convert_data_type.py [hunk 2]
```diff
@@ -31,7 +31,13 @@
...
     for out_type in TEST_NDARRAYS_ALL:
         TESTS_LIST.append(
-            ([in_type(np.array(1.0)), in_type(np.array(1.0))], out_type(np.array([1.0, 1.0])), True, None, False)  # type: ignore
+            (
+                [in_type(np.array(1.0)), in_type(np.array(1.0))],
+                out_type(np.array([1.0, 1.0])),
+                True,
+                None,
+                False,
+            )  # type: ignore
```
Comment (wyli) [inline tests/test_convert_data_type.py:40]: this change is a bug from the formatter... I'll revise

**Pull request merged**